FluentTerminal.SystemTray/Services/TerminalsManager.cs

```diff
@@ -8,21 +9,55 @@
 using System.Collections;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.IO;
 using System.Text;
+using System.Text.RegularExpressions;
 using Windows.ApplicationModel;
 
 namespace FluentTerminal.SystemTray.Services
 {
+    public struct TerminalSessionInfo
+    {
+        public DateTime StartTime { get; set; }
+        public string ProfileName { get; set; }
+        public ITerminalSession Session { get; set; }
+    }
+
     public class TerminalsManager
     {
-        private readonly Dictionary<byte, ITerminalSession> _terminals = new Dictionary<byte, ITerminalSession>();
+        private readonly Dictionary<byte, TerminalSessionInfo> _terminals = new Dictionary<byte, TerminalSessionInfo>();
 
         public event EventHandler<TerminalOutput> DisplayOutputRequested;
 
         public event EventHandler<TerminalExitStatus> TerminalExited;
 
+        private readonly ISettingsService _settingsService;
+
+        private static readonly Regex EscapeSequencePattern = new Regex(@"((\x9B|\x1B\[)[0-?]*[ -\/]*[@-~])|((\x9D|\x1B\]).*\x07)", RegexOptions.Compiled);
+
+        public TerminalsManager(ISettingsService settingsService)
+        {
+            _settingsService = settingsService;
+        }
+
         public void DisplayTerminalOutput(byte terminalId, byte[] output)
         {
+            if (_settingsService.GetApplicationSettings().EnableLogging && Directory.Exists(_settingsService.GetApplicationSettings().LogDirectoryPath))
+            {
+                var logOutput = output;
+                if (_settingsService.GetApplicationSettings().PrintableOutputOnly)
+                {
+                    string strOutput = System.Text.Encoding.UTF8.GetString(logOutput);
+                    strOutput = EscapeSequencePattern.Replace(strOutput, "");
+                    logOutput = Encoding.UTF8.GetBytes(strOutput);
+                }
+
+                var logFileStream = System.IO.File.Open(_settingsService.GetApplicationSettings().LogDirectoryPath + @"\" + 
+                    _terminals[terminalId].StartTime.ToString("yyyyMMddhhmmssfff") + "_" + _terminals[terminalId].ProfileName + ".log", System.IO.FileMode.Append);
+                logFileStream.Write(logOutput, 0, logOutput.Length);
+                logFileStream.Close();

```

Comment:
put this in a `finally` block to ensure the file is always closed?

Reply:
@mjs right, have put it into `using` + `try/catch`